lib/chef/knife/azure_server_create.rb

```diff
@@ -912,7 +909,26 @@ def get_chef_extension_public_params
 
       def get_chef_extension_private_params
         pri_config = Hash.new
-        pri_config[:validation_key] = File.read(Chef::Config[:validation_key])
+        if (Chef::Config[:validation_key] && !File.exist?(File.expand_path(Chef::Config[:validation_key])))
+
+          if Chef::VERSION.split('.').first.to_i == 11
+            ui.error('Unable to find validation key. Please verify your configuration file for validation_key config value.')
+            exit 1
+          end
+
+          # validator less bootstrap support for bootstrap protocol cloud-api
+          client_builder = Chef::Knife::Bootstrap::ClientBuilder.new(
+            chef_config: Chef::Config,
+            knife_config: config,
+            ui: ui,
+          )
+
+          client_builder.run
```

Comment:
There is already a rescue block in parent method create_server_def which calls this block so I don't think there is a need for another rescue block.